src/gui/GUIManager.ts

```diff
@@ -0,0 +1,676 @@
+    ["org.matrix.msc3215.abuse.nature.csam", "child sexual abuse material [likely illegal, consider warning authorities]"],
+    ["org.matrix.msc3215.abuse.nature.hate_speech", "spam"],
+    ["org.matrix.msc3215.abuse.nature.spam", "impersonation"],
+    ["org.matrix.msc3215.abuse.nature.impersonation", "impersonation"],
+    ["org.matrix.msc3215.abuse.nature.doxxing", "non-consensual sharing of identifiable private information of a third party (doxxing)"],
+    ["org.matrix.msc3215.abuse.nature.violence", "threats of violence or death, either to self or others"],
+    ["org.matrix.msc3215.abuse.nature.terrorism", "terrorism [likely illegal, consider warning authorities]"],
+    ["org.matrix.msc3215.abuse.nature.unwanted_sexual_advances", "unwanted sexual advances, sextortion, ... [possibly illegal, consider warning authorities]"],
+    ["org.matrix.msc3215.abuse.nature.ncii", "non consensual intimate imagery, including revenge porn"],
+    ["org.matrix.msc3215.abuse.nature.nsfw", "NSFW content (pornography, gore...) in a SFW room"],
+    ["org.matrix.msc3215.abuse.nature.disinformation", "disinformation"],
+]);
+const NATURE_DESCRIPTIONS = new Map(NATURE_DESCRIPTIONS_LIST);
+
+
+export enum Kind {
+    //! A MSC3215-style moderation request
+    MODERATION_REQUEST,
+    //! An abuse report, as per https://matrix.org/docs/spec/client_server/r0.6.1#post-matrix-client-r0-rooms-roomid-report-eventid
+    SERVER_ABUSE_REPORT,
+    //! Mjölnir encountered a problem while attempting to handle a moderation request or abuse report
+    ERROR,
+    //! A moderation request or server abuse report escalated by the server/room moderators.
+    ESCALATED_REPORT,
+    //! A message in the trashcan.
+    MESSAGE_IN_TRASHCAN,
+}
+
+/// The hardcoded `confirm` string, as embedded in confirmation reactions.
+const CONFIRM = "confirm";
+/// The hardcoded `cancel` string, as embedded in confirmation reactions.
+const CANCEL = "cancel";
+
+/**
+ * An abuse report received from a user.
+ *
+ * Note: These reports end up embedded in Matrix messages,
+ * so we're using Matrix naming conventions rather than JS/TS
+ * naming conventions.
+ */
+export interface IReport {
+    /**
+     * The user who sent the abuse report.
+     */
+    readonly accused_id: string,
+
+    /**
+     * The user who sent the message reported as abuse.
+     */
+    readonly reporter_id: string,
+
+    /**
+     * The room in which `eventId` took place.
+     */
+    readonly room_id: string,
+    readonly room_alias_or_id: string,
+
+    /**
+     * The event reported as abuse.
+     */
+    readonly event_id: string,
+}
+
+/**
+ * An abuse report, extended with the information we need for a confirmation report.
+ *
+ * Note: These reports end up embedded in Matrix messages, behind key `ABUSE_ACTION_CONFIRMATION_KEY`,
+ * so we're using Matrix naming conventions rather than JS/TS naming conventions.
+*/
+export interface IReportWithAction extends IReport {
```

Comment:
is this supposed to generalise the reaction based UI? then why would we have report stuff in this file?